content/blog/nuttx-getting-started/index.md

```diff
@@ -0,0 +1,280 @@
+---
+
+## Introduction to Apache NuttX RTOS
+
+According to the Apache NuttX website:
+<cite>[NuttX is a real-time operating system (RTOS) with an emphasis on standards compliance and small footprint. Scalable from 8-bit to 64-bit microcontroller environments, the primary governing standards in NuttX are Posix and ANSI standards.][1] </cite>
+
+NuttX is the 2nd most popular community-based RTOS (along with Zephyr in the 1st position):
+
+| Operating system | First commit | Governance    | License                    | Contributors | Pulse (jun10/2024) |
+|------------------|--------------|---------------|----------------------------|--------------|--------------------|
+| Zephyr           | 2014         | community     | Apache 2.0                 | 100+         | 942                |
+| NuttX            | 2007         | community     | Apache 2.0                 | 100+         | 135                |
+| RT-Thread        | 2009         | community     | Apache 2.0                 | 100+         | 67                 |
+| RIOT             | 2010         | community     | LGPL2.1                    | 100+         | 71                 |
+| Tyzen RT         | 2015         | Samsung       | Apache 2.0                 | 100+         | 36                 |
+| myNewt           | 2015         | Community     | Apache 2.0                 | 100+         | 25                 |
+| mbed OS          | 2013         | ARM           | Apache 2.0 or BSD-3 Clause | 100+         | 7                  |
+| FreeRTOS         | 2004         | Richard Barry | MIT                        | 100+         | 6                  |
+| Contiki-NG       | 2016         | community     | BSD-3 Clause               | 100+         | 4                  |
+| CMSIS-5          | 2016         | ARM           | Apache 2.0                 | 100+         | 0                  |
+| Azure-RTOS       | 2020         | Microsoft     | Microsoft Software License | 10+          | archived           |
+
+— <cite>[Table by Alin Jerpelea, presented on NuttX Workshop 2024][2]</cite>
+
+Its standards conformance (POSIX and ANSI, mostly) allows software developed under other OSes (under the same standards, such as software developed for Linux) to be easily ported to NuttX, enabling embedded applications to reuse tested applications. In such a sense, NuttX can be seen as the closest alternative to Linux for embedded software, providing interfaces similar to the ones used by Embedded Linux applications to SoCs that do not require running Linux directly.
+
+NuttX supports more than 300 boards from different architectures and its active community keeps providing support for newer devices and boards.Espressif SoCs are supported on NuttX!
+
+This tutorial shows how to install the required dependencies, including the toolchains to build Espressif SoCs, get NuttX's source code, configure a board (ESP32 DevKitC, in this example), build and flash NuttX into the ESP32-DevKitC board and connect to a Wi-Fi network.
```

Comment:
Now the introduction is buried among other paragraphs. Suggest placing it **in the beginning** of the article.

Also, the introduction should start not with installing required dependencies, but with key parts that users will take away with them: overview of NuttX in the context of Espressif SoCs, flashing NuttX and connecting to Wi-Fi,

How about something like this (feel free to modify anything):

In this tutorial, we will do a quick overview of NuttX and its compatibility with Espressfi SoCs. After that, we will build and flash NuttX onto the ESP32-DevKitC board and connect to a Wi-Fi network. Along the way, we will also install the required dependencies for NuttX, set up the toolchains to build applications for Espressif SoCs, get NuttX's source code, and configure the NuttX kernel before building.

Reply:
Good point! That's perfect...